src/inc/til/string.h

```diff
@@ -59,15 +61,110 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
         return str.size() >= prefix.size() && Traits::compare(str.data() + (str.size() - prefix.size()), prefix.data(), prefix.size()) == 0;
 #pragma warning(pop)
-    };
+    }
 
-    constexpr bool ends_with(const std::string_view str, const std::string_view prefix) noexcept
+    constexpr bool ends_with(const std::string_view& str, const std::string_view& prefix) noexcept
     {
         return ends_with<>(str, prefix);
-    };
+    }
 
-    constexpr bool ends_with(const std::wstring_view str, const std::wstring_view prefix) noexcept
+    constexpr bool ends_with(const std::wstring_view& str, const std::wstring_view& prefix) noexcept
     {
         return ends_with<>(str, prefix);
-    };
+    }
+
+    // Just like std::tolower, but without annoying locales.
+    template<typename T>
+    constexpr T tolower_ascii(T c)
+    {
+        if ((c >= 'A') && (c <= 'Z'))
+        {
+            c |= 0x20;
+        }
+
+        return c;
+    }
+
+    // Just like std::toupper, but without annoying locales.
+    template<typename T>
+    constexpr T toupper_ascii(T c)
+    {
+        if ((c >= 'a') && (c <= 'z'))
+        {
+            c &= ~0x20;
+        }
+
+        return c;
+    }
+
+    // Just like _memicmp, but without annoying locales.
+    template<typename T, typename Traits>
+    _TIL_INLINEPREFIX bool equals_insensitive_ascii(const std::basic_string_view<T, Traits>& str1, const std::basic_string_view<T, Traits>& str2) noexcept
+    {
+        if (str1.size() != str2.size())
+        {
+            return false;
+        }
+
+#pragma warning(push)
+#pragma warning(disable : 26429) // Symbol 'data1' is never tested for nullness, it can be marked as not_null
+#pragma warning(disable : 26481) // Don't use pointer arithmetic. Use span instead
+        auto remaining = str1.size();
+        auto data1 = str1.data();
+        auto data2 = str2.data();
+        for (; remaining; --remaining, ++data1, ++data2)
+        {
+            if (*data1 != *data2 && tolower_ascii(*data1) != tolower_ascii(*data2))
+            {
+                return false;
+            }
+        }
+#pragma warning(pop)
+
+        return true;
+    }
+
+    inline bool equals_insensitive_ascii(const std::string_view& str1, const std::string_view& str2) noexcept
+    {
+        return equals_insensitive_ascii<>(str1, str2);
+    }
+
+    inline bool equals_insensitive_ascii(const std::wstring_view& str1, const std::wstring_view& str2) noexcept
+    {
+        return equals_insensitive_ascii<>(str1, str2);
+    }
+
+    // Give the arguments ("foo bar baz", " "), this method will
+    // * modify the first argument to "bar baz"
+    // * return "foo"
+    // If the needle cannot be found the "str" argument is returned as is.
+    template<typename T, typename Traits>
+    _TIL_INLINEPREFIX std::basic_string_view<T, Traits> prefix_split(std::basic_string_view<T, Traits>& str, const std::basic_string_view<T, Traits>& needle) noexcept

```

Comment:
I _believe_ that since this is a template, it does not require `_TIL_INLINEPREFIX`. That's only for non-template functions that we want to emit only a single COMDAT-folded copy of.

```suggestion
    std::basic_string_view<T, Traits> prefix_split(std::basic_string_view<T, Traits>& str, const std::basic_string_view<T, Traits>& needle) noexcept
```